Docker/Component.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace Keboola\DockerBundle\Docker;
+
+use Keboola\Gelf\ServerFactory;
+use Keboola\Syrup\Exception\ApplicationException;
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+
+class Component
+{
+    /**
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @var array
+     */
+    private $data;
+
+    /**
+     * @var string
+     */
+    private $networkType;
+
+    /**
+     * Component constructor.
+     * @param array $componentData Component data as returned by Storage API
+     */
+    public function __construct(array $componentData)
+    {
+        $this->id = empty($componentData['id']) ? '' : $componentData['id'];
+        $data = empty($componentData['data']) ? [] : $componentData['data'];
+        try {
+            $this->data = (new Configuration\Component())->parse(['config' => $data]);
+        } catch (InvalidConfigurationException $e) {
+            throw new ApplicationException("Image definition is empty or invalid: " . $e->getMessage(), $e, $data);
+        }
+        $this->networkType = $this->data['network'];
+    }
+
+    /**
+     * @return string
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string
+     */
+    private function getSanitizedComponentId()
+    {
+        return preg_replace('/[^a-zA-Z0-9-]/i', '-', $this->getId());
+    }
+
+    /**
+     * @return string
+     */
+    public function getConfigurationFormat()
+    {
+        return $this->data['configuration_format'];
+    }
+
+    /**
+     * @return array
+     */
+    public function getImageParameters()
+    {
+        return $this->data['image_parameters'];
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasDefaultBucket()
+    {
+        return !empty($this->data['default_bucket']);
+    }
+
+    /**
+     * @param $configId
+     * @return string
+     */
+    public function getDefaultBucketName($configId)
+    {
+        return $this->data['default_bucket_stage'] . ".c-" . $this->getSanitizedComponentId() . "-" . $configId;
+    }
+
+    /**
+     * @return bool
+     */
+    public function injectEnvironment()
```

Comment:
http://softwareengineering.stackexchange.com/questions/232213/should-i-always-use-is-as-prefix-for-boolean-variables :)